dashboard/app/controllers/levels_controller.rb

```diff
@@ -203,6 +204,10 @@ def embed_level
     render 'levels/show'
   end
 
+  def set_applab_user_id
+    @applab_user_id = applab_user_id
```

Comment:
ok, calling applab_user_id directly does seem to work. The only argument I have for the existing pattern I've copied is that it avoids setting applab_user_id (thereby avoiding a database call for logged-in users) for actions besides 'show' and 'edit' . @laurelfan or @joshlory , my understanding of rails is shaky -- how often (or in what situations) are other actions besides 'show' and 'edit' called for the LevelsController and LevelSourcesController?